pyquil/quil.py

```diff
@@ -679,6 +821,9 @@ def __add__(self, other: InstructionDesignator) -> "Program":
         p = Program()
         p.inst(self)
         p.inst(other)
+        p._calibrations = self.calibrations
+        if isinstance(other, Program):
+            p.calibrations.extend(other.calibrations)
```

Comment:
This should also account for waveforms and frames. Without that, this is a bug.

Reply:
https://github.com/rigetti/pyquil/pull/1263